### ISSUE TITLE
Cloud Endpoints, python, UnboundLocalError: local variable 'scope_spec' referenced before assignment

### ISSUE THREAD

**Issue by myloginistaken** [google/apitools#274]
Using App Engine with Cloud Endpoints Framework for Python. In **apitools.base.py.credentials_lib.py** at l**ine 140** the variable **scope_spec** is referenced before assignment.
Full stacktrace:
```(/base/alloc/tmpfs/dynamic_runtimes/python27g/6a5167f0cae8960d/python27/python27_lib/versions/1/google/appengine/runtime/wsgi.py:269)

Traceback (most recent call last):

  File "/base/alloc/tmpfs/dynamic_runtimes/python27g/6a5167f0cae8960d/python27/python27_lib/versions/1/google/appengine/runtime/wsgi.py", line 240, in Handle
    handler = _config_handle.add_wsgi_middleware(self._LoadHandler())

  File "/base/alloc/tmpfs/dynamic_runtimes/python27g/6a5167f0cae8960d/python27/python27_lib/versions/1/google/appengine/runtime/wsgi.py", line 311, in _LoadHandler
    handler, path, err = LoadObject(self._handler)

  File "/base/alloc/tmpfs/dynamic_runtimes/python27g/6a5167f0cae8960d/python27/python27_lib/versions/1/google/appengine/runtime/wsgi.py", line 85, in LoadObject
    obj = __import__(path[0])

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/main.py", line 167, in <module>
    api = endpoints.api_server([AIMyDataAPI])

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints/apiserving.py", line 601, in api_server
    controller.start()

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints_management/control/client.py", line 253, in start
    self._initialize_flushing()

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints_management/control/client.py", line 372, in _initialize_flushing
    self._flush_schedule_report_aggregator()

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints_management/control/client.py", line 465, in _flush_schedule_report_aggregator
    transport = self._create_transport()

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints_management/control/client.py", line 151, in create_transport
    local.transport = _create_http_transport()

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints_management/control/client.py", line 143, in _create_http_transport
    log_response=do_logging)

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/endpoints_management/gen/servicecontrol_v1_client.py", line 50, in __init__
    additional_http_headers=additional_http_headers)

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/apitools/base/py/base_api.py", line 261, in __init__
    self._SetCredentials(**credentials_args)

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/apitools/base/py/base_api.py", line 309, in _SetCredentials
    self._credentials = credentials_lib.GetCredentials(**args)

  File "/base/data/home/apps/e~aimydata/20190626t111403.419174768539258548/lib/apitools/base/py/credentials_lib.py", line 140, in GetCredentials
    if isinstance(scope_spec, six.string_types):

UnboundLocalError: local variable 'scope_spec' referenced before assignment```

**Comment by houglum** [comment]
Ah, looks like a manual merge error from #271.  Apologies for not catching that during review.

@catleeball Let's pull in these changes and make sure everything still passes for the gsutil Py3 prerelease.

**Comment by houglum** [comment]
Released 0.5.30 with this fix, marking as closed.
